### DISCUSSION TITLE
Can mediamtx directly transmit local videos?

### DISCUSSION THREAD

**Discussion by Double-Lin** [bluenviron/mediamtx#3913] [Questions]
### Question

Can mediamtx directly transmit local videos?
I want to configure and directly transmit local videos to the client through on-demand mode instead of live streaming
thank you

**Comment by aler9** [comment]
If you want to serve an on-demand file that is synchronized among all the readers, instructions are [in the README](https://github.com/bluenviron/mediamtx?tab=readme-ov-file#on-demand-publishing).

If you want an independent stream for each reader, discard MediaMTX, put your video files (in MP4 format) in a folder and use nginx:

```
docker run --rm -it -v $PWD:/usr/share/nginx/html:ro -p 8080:80 -nginx
```

Serving recordings as .mp4 files with an HTTP server provides the best experience since it allows seeking, caching and is compatible with every client.
